app/routers/guest/applications.py

```diff
@@ -473,6 +473,66 @@ async def confirm_credit_product(
         )
 
 
+@router.post(
+    "/applications/rollback-confirm-credit-product",
+    tags=["applications"],
+)
+async def rollback_confirm_credit_product(
+    payload: parsers.ApplicationBase,
+    background_tasks: BackgroundTasks,
+    session: Session = Depends(get_db),
+    application: models.Application = Depends(
+        dependencies.get_scoped_application_as_guest_via_payload(
+            scopes=(dependencies.ApplicationScope.UNEXPIRED,), statuses=(models.ApplicationStatus.ACCEPTED,)
+        )
+    ),
+) -> serializers.ApplicationResponse:
+    """
+    Rollbascks the confirmation of the selected credit product for an application.
+
+    :param payload: The application data.
+    :return: The application response containing the updated application, borrower, award, lender, documents, and
+             credit product.
+    :raise: HTTPException with status code 400 if the credit product is not selected or not found.
+    """
+    with rollback_on_error(session):
+        if not application.credit_product_id:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Credit product not selected",
+            )
+
+        credit_product = models.CreditProduct.first_by(session, "id", application.credit_product_id)
+        if not credit_product:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Credit product not found",
+            )
+
+        application.lender_id = None
+        application.amount_requested = None
+        application.repayment_years = None
+        application.repayment_months = None
+        application.payment_start_date = None
+        application.pending_documents = False
```

Comment:
@nahu I think we should also need to set:
``` python
application.credit_product_id = None
application.calculator_data = {}
application.borrower_credit_product_selected_at = None
```
and :

``` python
        models.ApplicationAction.create(
            session,
            type=models.ApplicationActionType.APPLICATION_CALCULATOR_DATA_UPDATE,
            data=jsonable_encoder(payload, exclude_unset=True),
            application_id=application.id,
        )
```

With a new action type e.g. APPLICATION_ROLLBACK_CREDIT_PRODUCT